integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/massindexing/MassIndexingFailureIT.java

```diff
@@ -529,6 +526,21 @@ private void doMassIndexingWithFailure(SessionFactory sessionFactory,
 			Runnable ... expectationSetters) {
 		doMassIndexingWithFailure(
 				sessionFactory,
+				searchSession -> searchSession.massIndexer(),
+				threadExpectation,
+				thrownExpectation,
+				expectationSetters
+		);
+	}
+
+	private void doMassIndexingWithFailure(SessionFactory sessionFactory,
```

Comment:
IMHO too many indirections on helper methods make the test code less readable. If it is possible, I would not create a new helper method here.

Reply:
While I agree indirection should be avoided if possible, here the alternative is copy/pasting 70 lines of code into a dozen different tests. That would hardly be more readable, and certainly not more maintainable.

That being said, I made even more changes to this test in another PR, and I suspect you won't like these changes either. Let's discuss this there, and I'll consider changing this if you can provide a reasonable alternative.

Reply:
I think there is something in the middle between avoid indirection at all and the current solution.
I would reorganize the sub-calls so that we do not never reach a depth of 3.
Basically one main method that every test method eventually would use, using an optional  indirection method at most to reach it.
This comment was a nice to have, it is not critical.